addons/crm/models/crm_lead.py

```diff
@@ -306,10 +306,9 @@ def _onchange_compute_probability(self, optional_field_name=None):
         if optional_field_name and optional_field_name not in self._pls_get_safe_fields():
             return
         lead_probabilities = self._pls_get_naive_bayes_probabilities()
-        if self.id in lead_probabilities:
-            self.automated_probability = lead_probabilities[self.id]
-            if self._origin.is_automated_probability:
-                self.probability = self.automated_probability
+        self.automated_probability = lead_probabilities.get(self.id, 0)
```

Comment:
Behavior still feels a bit weird on a runbot.

I have this lead, probability to win is 30% in "proposition" stage:
![Selection_465](https://user-images.githubusercontent.com/41282536/78637274-5a1f5e00-78aa-11ea-91fe-7eff5c42851e.png)

I set it as "won", correctly goes up to 100%, so far so good:
![Selection_466](https://user-images.githubusercontent.com/41282536/78637273-5986c780-78aa-11ea-9217-c9cddb3657ee.png)

But then, if I say "woops, I misclicked, let's put it in proposition stage again":
![Selection_467](https://user-images.githubusercontent.com/41282536/78637270-58ee3100-78aa-11ea-9f06-f65802c8246f.png)

Probability to win is now 95.45% ??
That doesn't feel right.

Then about the code itself, I feel like defaulting to 0 is maybe not the most logical thing to do, why not False instead?
0 means "there is no way you're going to win this"
while False is more "I don't know".

And finally, I don't understand the commit message too much, especially the last sentence.

Reply:
When setting the lead to stage won, proba and auto proba are set to 100%
As in won stage, every lead are won.. to probability to win 100% (thanks Captain Obvious!)
When setting back to another stage, as the probabilities were aligned, when recomputing the auto-probability, we also align the probability.
So this is quite logic. If the lead was originally in manual mode, it will end up in auto mode.
This is known limitation.

Reply:
Note that if you change the stage in edit mode, you will keep your manual probability until you save.